### PR TITLE
ACT: Support `Context Info` action

### PR DESCRIPTION
**Pull request by mchernyavsky**
Refactoring for RsStructureViewModel.
Support for nested items in the structure view model of a function.
Implementation of DeclarationRangeHandler.
The `View | Context Info` feature is provided by the platform using StructureViewModel and DeclarationRangeHandler implementations.

### CODE REVIEW

File: src/main/kotlin/org/rust/ide/structure/RsStructureViewModel.kt [hunk 1]
```diff
@@ -44,44 +45,52 @@ private class RsStructureViewElement(
...
+            for (item in psi.itemsAndMacros) {
+                when (item) {
+                    is RsMacro,
+                    is RsFunction,
+                    is RsModDeclItem,
+                    is RsModItem,
+                    is RsStructOrEnumItemElement,
+                    is RsTraitOrImpl,
+                    is RsTypeAlias,
+                    is RsConstant -> add(item)
```
Comment (fedochet) [inline src/main/kotlin/org/rust/ide/structure/RsStructureViewModel.kt]: Make a newline to for better visual separation

**Comment by mchernyavsky** [comment]
bors retry

**Comment by ortem** [comment]
bors retry

**Comment by mchernyavsky** [comment]
bors retry

**Comment by Undin** [comment]
bors r=ortem

**Pull request merged**